scipy/stats/_hypotests.py

```diff
@@ -112,17 +112,13 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8), *, axis=0):
     np = array_namespace(x, y)
     # x and y are converted to arrays by the decorator
     # and `axis` is guaranteed to be -1.
-    t = np.asarray(t)
+    x, y = xp_promote(x, y, force_floating=True, xp=np)
+    t = np.asarray(t, dtype=x.dtype)
```

Comment:
Not worth rerunning CI for but if we are coming back here in the near future then
```suggestion
    x, y, t = xp_promote(x, y, t, force_floating=True, xp=np)
```

Reply:
I wrote it this way intentionally because `t` is not necessarily an array. Even if it is an array, it is more like a sequence of separate floats. I didn't think we wrote `xp_promote` to accept non-array array-likes, and we don't want to coerce it to an array before dtypes determination because that could upcast everything to the default float.